### PR TITLE
Implement file based coredump thread for k8s/docker nodes

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/pOdW1Bm4/2252-scylla-operator-coredumps
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by bentsi** [comment]
Please resolve conflicts

**Comment by dkropachev** [comment]
> Please resolve conflicts

Done.

aws test run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/404/
minikube test run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/79

**Comment by dkropachev** [comment]
Tried to satisfy deepcode with stupid __ne__ issue, comments does not work, adding __ne__ to SctEvent does not help to stop having this issue alerted on SctEvent's descedants.

File: sdcm/coredump.py [hunk 1]
```diff
@@ -53,31 +55,49 @@ def __str__(self):
             return f'CoreDump[{self.pid}, {self.corefile}]'
         return f'CoreDump[{self.pid}]'
 
+    def update(self,
```
Comment (bentsi) [inline sdcm/coredump.py:58]: why do we need this?

Comment (dkropachev) [reply sdcm/coredump.py:58]: before this change we created coredumpInfo entity at the end, now we create it at the begging and add data to it once it is possible.

**Review by bentsi — APPROVED** [review]

**Pull request merged**